pkg/resource/template/resource.go

```diff
@@ -134,7 +134,7 @@ func (t *TemplateResource) setVars() error {
 // template and setting the desired owner, group, and mode. It also sets the
 // StageFile for the template resource.
 // It returns an error if any.
-func (t *TemplateResource) createStageFile() error {
+func (t *TemplateResource) createStageFile() (err error) {
```

Comment:
One more change that isn't required.  Reason I'm not fond of this change is because it's confusing thinking about global and local scope of the same name... easier to avoid the confusion :-)